src/vm/tieredcompilation.cpp

```diff
@@ -710,6 +711,8 @@ void TieredCompilationManager::ActivateCodeVersion(NativeCodeVersion nativeCodeV
     // code version will activate then.
     ILCodeVersion ilParent;
     HRESULT hr = S_OK;
+    MethodDescBackpatchInfoTracker::ConditionalLockHolder lockHolder;
```

Comment:
Could move this to just before the two places where the `TableLockHolder` is instantiated to decrease the lock scope

Reply:
This lock needs to be taken outside of any thread suspension activities, I could move it inside the scope, but it wouldn't be in a different location.